docs/operate-a-stake-pool/hardening-server.md

```diff
@@ -0,0 +1,310 @@
+---
+id: hardening-server
+title: Hardening the server
+sidebar_label: Hardening the server
+description: "Stake pool guide: Learn how to harden your server"
+image: ../img/og/og-developer-portal.png
+---
+
+Before installing your Cardano node, you need to have a clean and secured OS. This guide explains how to secure your server, in 10 steps.
+
+### Prerequisites 
+
+- This guide is designed for Ubuntu Server (22.04 LTS or 20.04 LTS)
+- A SSH client on your local machine. If you are using a Windows machine, install OpenSSH Client ([guide](https://www.howtogeek.com/311287/how-to-connect-to-an-ssh-server-from-windows-macos-or-linux/))
+- SSH server up and running on your Ubuntu Server
+
+### 1- Create a non-root user for your Cardano node
+
+First thing first : you should never use the root account to connect to your server and manage it. Instead, use a non-root account, that can use sudo. We are going to create one.
+
+- Connect to your server with SSH (for this time only, you might use root. We'll disable it after anyway. You can also use the default user created by the Ubuntu installer).
+
+```shell
+ssh <defaultuser>@server-ip
+```
+
+- New user creation called cardano (or whatever you want)
+
+```shell
+sudo useradd -m -s /bin/bash cardano
+```
+
+- Set a strong password
+```shell
+sudo passwd cardano
+```
+
+- Allow cardano user to use sudo command
+```shell
+sudo usermod -aG sudo cardano
+```
+
+- Disconnect from your server, and reconnect using your new user
+```shell
+ssh cardano@server-ip
+```
+
+- Remove default user if needed
+```shell
+sudo userdel <defaultuser>
+```
+
+### 2- Disable root
+
+You should never use root account. Always use sudo to run commands with privileges. We are going to disable it.
+
+```shell
+sudo passwd -l root
+```
+
+### 3- Update System
+
+It's time to update your system with the latest pathes available.
+
+```shell
+sudo apt-get update -y
+sudo apt-get upgrade -y
+sudo apt-get autoremove
+sudo apt-get autoclean
+```
+
+Reboot your server.
+```shell
+sudo reboot
+```
+
+### 4- Activate Unattended-upgrades for automatic security updates
+
+Next step is to enable unattended upgrades to automatically install security updates (this is the default behavior of unattended-upgrades)
+
+```shell
+sudo apt-get install unattended-upgrades
```

Comment:
This can be a bit risky for node, as you could end up with  dependent runtime library/package being updated (in future)

Reply:
While technically true, i believe unattended-upgrades for security updates is a must have. Without it, you are taking the risk to never update packages, and host software with remotely exploitable security vulnerabilities. It all comes down to risks versus benefits. I could add an "optional" label or an admonition text ?

Reply:
I added an admonition text for the unattended-upgrades